src/chronify/time_series_mapper_index.py

```diff
@@ -0,0 +1,268 @@
+        from_schema_kwargs["time_config"] = time_config
+
+        int_schema = TableSchema(**from_schema_kwargs)
+        return int_schema
+
+    def map_time(
+        self,
+        check_mapped_timestamps: bool = False,
+        **kwargs,
+    ) -> None:
+        """Convert time columns with from_schema to to_schema configuration."""
+        self.check_schema_consistency()
+
+        map_table_name = "map_table"
+        dfm = self._create_mapping()
+        mapping_schema = self._create_mapping_schema(map_table_name)
+
+        # if using datetime mapper, check timestamps in that mapper
+        tmp_check_mapped_timestamps = self._datetime_mapper is None
+
+        apply_mapping(
+            dfm,
+            mapping_schema,
+            self._from_schema,
+            self._to_schema,
+            self._engine,
+            self._metadata,
+            check_mapped_timestamps=tmp_check_mapped_timestamps,
+            **kwargs,
+        )
+
+        if self._datetime_mapper is not None:
+            self._datetime_mapper.map_time(
+                check_mapped_timestamps=check_mapped_timestamps, **kwargs
+            )
+
+            # TODO drop table with self._to_schema.name
+
+    def create_single_tz_mapping_dataframe(
+        self, to_time_col: str, from_time_col: str
+    ) -> pd.DataFrame:
+        """
+        Creates a dataframe mapping by subtracting timestamps from the start_timestamp of
+        the index_time schema.
+
+        timestamps and start_timestamp need to either both be timezone aware or both be timezone naive
+        """
+
+        dfm = pd.DataFrame(
+            {
+                from_time_col: self._index_generator.list_timestamps(),
+                to_time_col: self._datetime_generator.list_timestamps(),
+            }
+        )
+
+        if self._align_local_clocktime:
+            df_idx_clock_time = pd.DataFrame(
+                {
+                    from_time_col: self._index_generator.list_timestamps(),
+                    "clock_time": pd.date_range(
+                        start=self._from_time_config.start_timestamp,  # should be naive
+                        periods=self._from_time_config.length,
+                        freq=self._from_time_config.resolution,
+                    ),
+                }
+            )
+
+            dfm["clock_time"] = dfm[to_time_col].dt.tz_localize(None)
+            dfm = dfm.drop(from_time_col, axis=1)
+            dfm = dfm.merge(df_idx_clock_time, on="clock_time")
+
+        return dfm
+
+    def create_multi_tz_aligned_mapping_dataframe(
+        self, to_time_col: str, from_time_col: str, time_zones: list[str]
+    ) -> pd.DataFrame:
+        # Create mapping using from/to schema
+        dfm = self.create_single_tz_mapping_dataframe(to_time_col, from_time_col)
+        df_mtz = []
+        for tz in time_zones:
+            dfc = dfm.copy()
+            dfc[to_time_col] = dfc[to_time_col].dt.tz_convert(tz)
```

Comment:
Sorry if I am not following this correctly. But if `self._align_local_clocktime, dfm[to_time_col]` would be tz-naive right? Would tz_convert() even work here?